File: app/src/remote/java/org/mesonet/app/forecast/FiveDayForecastDataController.java
package org.mesonet.app.forecast;


import org.mesonet.app.DataDownloader;
import org.mesonet.app.site.mesonetdata.MesonetSiteDataController;

import java.util.Observable;
import java.util.Observer;

import javax.inject.Inject;

public class FiveDayForecastDataController extends BaseFiveDayForecastDataController implements Observer
{
    DataDownloader mDataDownloader;

    MesonetSiteDataController mMesonetSiteDataController;


    @Inject
    public FiveDayForecastDataController(MesonetSiteDataController inSiteDataController, DataDownloader inDownloader)
    {
        mDataDownloader = inDownloader;
        mMesonetSiteDataController = inSiteDataController;
        Update();
    }



    private void Update()
    {
        if(!mMesonetSiteDataController.SiteDataFound())
        {
            mMesonetSiteDataController.addObserver(this);
            return;
        }
        mDataDownloader.Download("http://www.mesonet.org/index.php/app/forecast/" + mMesonetSiteDataController.CurrentSelection(), 0, new DataDownloader.DownloadCallback() {
            @Override
            public void DownloadComplete(int inResponseCode, String inResult) {
                SetData(inResult);
            }

            @Override
            public void DownloadFailed() {

            }
        }, new DataDownloader.DownloadUpdateCheckListener() {
            @Override
            public void CheckForUpdate(long inLastUpdated) {

            }

            @Override
            public boolean ContinueUpdates() {
                return false;
            }

            @Override
            public long GetInterval() {
                return 60000;
            }
        });
    }

    @Override
    public void update(Observable o, Object arg) {
        Update();
    }
}

File: app/src/remote/java/org/mesonet/app/forecast/FiveDayForecastDataController.kt
package org.mesonet.app.forecast


import org.mesonet.app.DataDownloader
import org.mesonet.app.site.mesonetdata.MesonetSiteDataController

import java.util.Observable
import java.util.Observer

import javax.inject.Inject

class FiveDayForecastDataController @Inject
constructor(internal var mMesonetSiteDataController: MesonetSiteDataController, internal var mDataDownloader: DataDownloader) : BaseFiveDayForecastDataController(), Observer {


    init {
        Update()
    }


    private fun Update() {
        if (!mMesonetSiteDataController.SiteDataFound()) {
            mMesonetSiteDataController.addObserver(this)
            return
        }
        mDataDownloader.Download("http://www.mesonet.org/index.php/app/forecast/" + mMesonetSiteDataController.CurrentSelection(), 0, object : DataDownloader.DownloadCallback {
            override fun DownloadComplete(inResponseCode: Int, inResult: String?) {
                SetData(inResult)
            }

            override fun DownloadFailed() {

            }
        }, object : DataDownloader.DownloadUpdateCheckListener {
            override fun CheckForUpdate(inLastUpdated: Long) {

            }

            override fun ContinueUpdates(): Boolean {
                return false
            }

            override fun GetInterval(): Long {
                return 60000
            }
        })
    }

    override fun update(o: Observable, arg: Any?) {
        Update()
    }
}
